frontend/src/hocs/withScreencast/withScreencast.tsx

```diff
@@ -29,31 +29,39 @@ import {
 
 import "./withScreencast.scss"
 
+export enum Steps {
+  UNSUPPORTED = "unsupported",
+  OFF = "off",
+  SETUP = "setup",
+  COUNTDOWN = "countdown",
+  RECORDING = "recording",
+  PREVIEW_FILE = "preview_file",
+}
+
 interface WithScreenCastProps {}
 
 interface WithScreenCastState {
   fileName: string
-  recording: boolean
   recordAudio: boolean
   countdown: number
   startAnimation: boolean
-  blob?: Blob
-  showRecordedDialog: boolean
-  showScreencastDialog: boolean
-  showUnsupportedDialog: boolean
+  outputBlob?: Blob
+  currentState: Steps
 }
 
-export interface ScreenCastHOC extends WithScreenCastState {
+export interface ScreenCastHOC {
+  currentState: Steps
   toggleRecordAudio: () => void
   startRecording: (fileName: string) => void
   stopRecording: () => void
 }
 
+// Supported browsers taken from https://caniuse.com/#search=getdisplaymedia
 const SUPPORTED_BROWSERS: any = {
-  Chrome: "49",
-  Firefox: "29",
-  Opera: "36",
-  Edge: "76",
+  Chrome: "72",
+  Firefox: "66",
+  Opera: "64",
+  Edge: "17",
```

Comment:
My other comment disappeared: https://github.com/streamlit/streamlit/pull/929#discussion_r369329620

I'm still wondering what happens in other browsers. Does Brave, for example, get detected as Chrome?

Reply:
I’ll test it just to make sure

Reply:
I tested Brave and it is returning Chrome

<img width="208" alt="Screen Shot 2020-01-23 at 10 08 31 AM" src="https://user-images.githubusercontent.com/631997/72987065-5a8a7b80-3dc8-11ea-9835-80a45b941a77.png">